File: file-fixed-sequential-write.pl
use strict;
use warnings;

use IO::File;

my $f = new IO::File 'bench.bin', '>' or die "Cannot open";
my $total = 1024 * 1024 * 1120;

my $buf = 'Q' x (1024 * 1024 * 8);
my $written = 0;

while ($written < $total)
{
	$written +=	$f->syswrite($buf) || die "Cannot write"; 
}

print "$written\n";

File: file-fixed-sequential-write.js
'use strict';

const SIZE = 8 * 1024 * 1024;
const fs = require('fs');
const buf = new Buffer(SIZE).fill('Q');
let total = 30 * 1024 * 1024 * 1024;

const ws = fs.createWriteStream('bench.bin');

const cb = err => {
  if (err) {
    console.log('err');
  }

  if (total <= 0) {
    return;
  }

  if (total <= SIZE) {
    const w = total;
    total = 0;
    ws.end(buf.slice(0, w), cb);
  } else {
    total -= SIZE;
    ws.write(buf, cb);
  }
};

console.log(total);
cb();
